node-tree-sitter.d.ts

```diff
@@ -0,0 +1,91 @@
+declare module "tree-sitter" {
+  export class Document {
+    // Getters
+    public rootNode: AstNode | null;
+
+    // Methods
+    public setLanguage(module: any): void;
+    public setInputString(input: string): void;
+    public parse(): void;
+    public getLogger(): LoggerFunc;
+    public setLogger(logFunc: LoggerFunc): void;
+    public getInput(): StringInput | null;
+    public setInput(input: StringInput): void;
+    public edit(delta: EditDelta): Document;
+    public invalidate(): void;
+  }
+
+  type LoggerFunc = (message: string, params: {[param: string]: string}, type: "parse" | "lex") => void;
+
+  type EditDelta = {
+    startIndex: number;
+    lengthRemoved: number;
+    lengthAdded: number;
+    startPosition: Position;
+    extentRemoved: Position;
+    extentAdded: Position;
+  };
+
+  export type Position = {
+    row: number;
+    column: number;
+  };
+
+  export type StringInput = {
+    position: number;
+    string: any;
+    bufferSize: number | null;
+    seek(distance: number): void;
+    read(): any;
+  }
```

Comment:
I think that `StringInput` should just be called `Input`, and it should be typed as follows:

```ts
export interface Input = {
  seek(index: number, position: Position): void;
  read(): String;
};
```

The only required methods are `seek` and `read`; all these other properties were just implementation details used in tests or in particular implementations (e.g. `StringInput`).

Reply:
In the code I only see seek taking in a single number argument, should it just be `seek(index: number): void;`?

Reply:
Oh yeah, that's right. The `position` argument was something that I planned to add but never did.